sw/device/lib/dif/dif_otbn.h

```diff
@@ -73,29 +73,37 @@ typedef enum dif_otbn_result {
 } dif_otbn_result_t;
 
 /**
- * Error codes received from the hardware.
+ * OTBN Errors
+ *
+ * OTBN uses a bitfield to indicate which errors have been seen. Multiple errors
+ * can be seen at the same time. This enum gives the indices of the different
+ * bits that may be set in the error bitfield.
  */
-typedef enum dif_otbn_err_code {
-  // Keep error codes in sync with the hardware!
-  /** No error occurred. */
-  kDifOtbnErrCodeNoError = 0x0,
+typedef enum dif_otbn_err_bit {
   /** Load or store to invalid address. */
-  kDifOtbnErrCodeBadDataAddr = 0x1,
+  kDifOtbnErrBitBadDataAddr = 0,
   /** Instruction fetch from invalid address. */
-  kDifOtbnErrCodeBadInsnAddr = 0x2,
+  kDifOtbnErrBitBadInsnAddr = 1,
   /** Call stack underflow or overflow. */
-  kDifOtbnErrCodeCallStack = 0x3,
+  kDifOtbnErrBitCallStack = 2,
   /** Illegal instruction execution attempted */
-  kDifOtbnErrCodeIllegalInsn = 0x4,
+  kDifOtbnErrBitIllegalInsn = 3,
   /** LOOP[I] related error */
-  kDifOtbnErrCodeLoop = 0x5,
+  kDifOtbnErrBitLoop = 4,
   /** Error seen in Imem read */
-  kDifOtbnErrCodeFatalImem = 0x80,
+  kDifOtbnErrBitFatalImem = 5,
   /** Error seen in Dmem read */
-  kDifOtbnErrCodeFatalDmem = 0x81,
-  /** Error seen in RF */
-  kDifOtbnErrCodeFatalReg = 0x82
-} dif_otbn_err_code_t;
+  kDifOtbnErrBitFatalDmem = 6,
+  /** Error seen in RF read */
+  kDifOtbnErrBitFatalReg = 7
+} dif_otbn_err_bit_t;
```

Comment:
Most (all?) software uses will want to use AND/OR to check for or set a bit; doing so is easier if we don't define the bit indices here, but the value (`1 << bit`).

Reply:
IMO, a sparse enum would do the trick just fine in my mind and also covers the "no error occurred" case reasonably nicely (can of course be combined with an enum for the bits).

```
typedef enum dif_otbn_err_code {
  kDifOtbnErrCodeNoError = 0x0,
  kDifOtbnErrCodeBadDataAddr = (1 << 0),
  kDifOtbnErrCodeBadInsnAddr = (1 << 1),
  kDifOtbnErrCodeCallStack = (1 << 2),
} dif_otbn_err_code_t;
```

But I don't care ultimately, whatever SW is happy with.

Reply:
Looks like I got the wrong end of the stick here and using an enum with the actual bits rather than the bit indices is perfectly fine in a DIF so I'll switch to using that.

Reply:
Sorry about that, I should have been clearer as to when the bit indexes were necessary, and when a bitfield was fine.